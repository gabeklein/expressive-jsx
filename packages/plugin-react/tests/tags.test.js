const test = require("./_adapter");

test("strip tag; emit default", `
  () => do {
    <hello> Hello World </hello>
  }
`);

test("pass-though capital (component) tags", `
  () => do {
    <Hello> Hello World </Hello>
  }
`);

test("pass-through HTML5 tags", `
  () => do {
    <div>
      <uncommon> ... </uncommon>
      <article> ... </article>
      <blockquote> ... </blockquote>
      <input> ... </input>
      <h1> ... </h1>
      <h2> ... </h2>
      <h3> ... </h3>
      <h4> ... </h4>
      <h5> ... </h5>
      <h6> ... </h6>
      <p> ... </p>
      <a> ... </a>
      <ul> ... </ul>
      <ol> ... </ol>
      <li> ... </li>
      <i> ... </i>
      <b> ... </b>
      <em> ... </em>
      <strong> ... </strong>
      <span> ... </span>
      <hr> ... </hr>
      <img> ... </img>
      <div> ... </div>
      <br> ... </br>
    </div>
  }
`);

test("Handle member-type tags", `
  () => do {
    <Foo.Bar />
  }
`);

test("Member property implemented as name", `
  Bar: {
    color: blue;
  }

  () => do {
    <Foo.Bar />
  }
`);