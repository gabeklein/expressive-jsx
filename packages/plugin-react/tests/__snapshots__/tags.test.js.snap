// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handle member-type tags 1`] = `"() => create(Foo.Bar, {});"`;

exports[`Handle member-type tags 2`] = `"() => <Foo.Bar />;"`;

exports[`Member property implemented as name 1`] = `
"() => create(Foo.Bar, {
  \\"className\\": \\"Bar_796\\"
});

Styles.include(\`
	.Bar_796 { color: blue }
\`);"
`;

exports[`Member property implemented as name 2`] = `
"() => <Foo.Bar className=\\"Bar_796\\" />;

Styles.include(\`
	.Bar_796 { color: blue }
\`);"
`;

exports[`pass-though capital (component) tags 1`] = `"() => create(Hello, {}, \\"Hello World\\");"`;

exports[`pass-though capital (component) tags 2`] = `"() => <Hello>Hello World</Hello>;"`;

exports[`pass-through HTML5 tags 1`] = `"() => create(\\"div\\", {}, create(\\"div\\", {}), create(\\"article\\", {}), create(\\"blockquote\\", {}), create(\\"input\\", {}), create(\\"h1\\", {}), create(\\"h2\\", {}), create(\\"h3\\", {}), create(\\"h4\\", {}), create(\\"h5\\", {}), create(\\"h6\\", {}), create(\\"p\\", {}), create(\\"a\\", {}), create(\\"ul\\", {}), create(\\"ol\\", {}), create(\\"li\\", {}), create(\\"i\\", {}), create(\\"b\\", {}), create(\\"em\\", {}), create(\\"strong\\", {}), create(\\"span\\", {}), create(\\"hr\\", {}), create(\\"img\\", {}), create(\\"div\\", {}), create(\\"br\\", {}));"`;

exports[`pass-through HTML5 tags 2`] = `"() => <div><div /><article /><blockquote /><input /><h1 /><h2 /><h3 /><h4 /><h5 /><h6 /><p /><a /><ul /><ol /><li /><i /><b /><em /><strong /><span /><hr /><img /><div /><br /></div>;"`;

exports[`strip tag; emit default 1`] = `"() => create(\\"div\\", {}, \\"Hello World\\");"`;

exports[`strip tag; emit default 2`] = `"() => <div>Hello World</div>;"`;
